src/main/java/codeu/model/data/User.java

```diff
@@ -118,6 +127,78 @@ public void setAdmin(Boolean admin) {
         this.admin = admin;
     }
 
+    /**
+     * Sets the pals of this User.
+     */
+    public void setPals(ArrayList<String> pals) {
+        this.pals = pals;
+    }
+
+    /**
+     * Returns the pals of this User.
+     */
+    public ArrayList<String> getPals() {
+        return pals;
+    }
+
+    /** Adds a pal to the list of this User's pals. */
+    public void addPal(String name) {
+        this.pals.add(name);
+    }
+
+    /** Checks whether a given name is in this User's list of pals. */
+    public boolean isPal(String name) {
+        return this.pals.contains(name);
+    }
+
+    /**
+     * Sets the incoming requests of this User.
+     */
+    public void setIncomingRequests(ArrayList<String> incoming) {
```

Comment:
same here.. always validate input of public methods

Reply:
oh yes, that's important thank you! I just added the condition to check